containers/Connector/Connector.tsx

```diff
@@ -71,52 +74,57 @@ const useConnector = () => {
     []
   );
 
-  const updateState = useCallback((update: AppState) => {
-    if (update.wallets.length > 0) {
-      const wallet = update.wallets[0].accounts[0];
+  const updateState = useCallback(
+    (update: AppState) => {
+      if (update.wallets.length > 0) {
+        const wallet = update.wallets[0].accounts[0];
 
-      const { label } = update.wallets[0];
-      const { id } = update.wallets[0].chains[0];
-      const networkId = getNetworkIdFromHex(id);
+        const { label } = update.wallets[0];
+        const { id } = update.wallets[0].chains[0];
+        const networkId = getNetworkIdFromHex(id);
 
-      const isSupported = isSupportedNetworkId(networkId) && isSupportedWalletChain(networkId);
+        const isSupported = isSupportedNetworkId(networkId) && isSupportedWalletChain(networkId);
 
-      const network = {
-        id: networkId,
-        name: isSupportedNetworkId(networkId) ? NetworkNameById[networkId] : 'Unsupported Network',
-        useOvm: getIsOVM(networkId),
-      };
+        const network = {
+          id: networkId,
+          name: isSupportedNetworkId(networkId)
+            ? NetworkNameById[networkId]
+            : 'Unsupported Network',
+          useOvm: getIsOVM(networkId),
+        };
 
-      const provider = new ethers.providers.Web3Provider(update.wallets[0].provider, {
-        name: network.name,
-        chainId: networkId,
-      });
-
-      const signer = provider.getSigner();
-      const useOvm = getIsOVM(Number(networkId));
-      const synthetixjs = isSupported ? synthetix({ provider, networkId, useOvm }) : null;
-
-      dispatch({
-        type: AppEvents.CONFIG_UPDATE,
-        payload: {
-          address: wallet.address,
-          walletWatched: null,
-          walletType: label,
-          network,
-          provider,
-          signer,
-          synthetixjs,
-          ensName: wallet?.ens?.name || null,
-          ensAvatar: wallet?.ens?.avatar?.url || null,
-        },
-      });
-
-      const connectedWallets = update.wallets.map(({ label }) => label);
-      localStorage.setItem(LOCAL_STORAGE_KEYS.SELECTED_WALLET, JSON.stringify(connectedWallets));
-    } else {
-      dispatch({ type: AppEvents.WALLET_DISCONNECTED });
-    }
-  }, []);
+        const provider = new ethers.providers.Web3Provider(update.wallets[0].provider, {
+          name: network.name,
+          chainId: networkId,
+        });
+
+        const signer = provider.getSigner();
+        const contracts = setSynthetix(networkId, signer);
+        const synthetixjs = isSupported ? ({ contracts } as SynthetixJS) : null;
```

Comment:
Here we set `synthetixjs` to just be the `{contracts}`. I played around with the types a little (removed `any`) and noticed that just a few lines before we try to create a  `synthsMap` and  `tokensMap` by accessing `synthetixjs.synths` and `synthetixjs.tokens`. These would now be undefined, right?

Reply:
Yep good pickup

Reply:
Merged fixes to resolve this